pyproject.toml

```diff
@@ -1,3 +1,114 @@
+[build-system]
+requires = ["setuptools>=61.2", "setuptools_scm[toml]>=3.4.3"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "trio"
+authors = [{name = "Nathaniel J. Smith", email = "njs@pobox.com"}]
+license = {text = "MIT OR Apache-2.0"}
+description = "A friendly Python library for async concurrency and I/O"
+keywords = ["async", "io", "networking", "trio"]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: MIT License",
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: POSIX :: Linux",
+    "Operating System :: MacOS :: MacOS X",
+    "Operating System :: POSIX :: BSD",
+    "Operating System :: Microsoft :: Windows",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: Implementation :: PyPy",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: System :: Networking",
+    "Framework :: Trio",
+]
+urls = {Homepage = "https://github.com/python-trio/trio"}
+requires-python = ">=3.7"
+dependencies = [
+    "attrs >= 19.2.0",
+    "sortedcontainers",
+    "async_generator >= 1.9",
+    "idna",
+    "outcome",
+    "sniffio",
+    # cffi 1.12 adds from_buffer(require_writable=True) and ffi.release()
+    # cffi 1.14 fixes memory leak inside ffi.getwinerror()
+    # cffi is required on Windows, except on PyPy where it is built-in
+    "cffi>=1.14; os_name == 'nt' and implementation_name != 'pypy'",
+    "exceptiongroup >= 1.0.0rc9; python_version < '3.11'",
+]
+dynamic = ["version"]
+
+[project.readme]
```

Comment:
I realize you just copied what setup.py did but if we're going to switch to PEP 621 we should use the opportunity to use README.md directly. But this requires making sure it renders well both in GitHub and PyPI.